public/js/post/post_page.js

```diff
@@ -0,0 +1,113 @@
+import {CourseCategory, MultipleCourseCode, InquiryCategory} from 'es6!post/course_components';
+import {SpecialistPost, MajorPost, MinorPost} from 'es6!post/post_components';
+
+var CheckMyPost = React.createClass({
+
+    componentDidMount: function() {
+        var activeTab = this.refs.postNav.getActiveTab() + 'Post';
+        this.changeActiveTab(activeTab);
+        this.updateNavCreditCounts();
+    },
+
+    changeActiveTab: function(newTab) {
+        var activeTab = newTab;
+        var tabs = ['spePost', 'majPost', 'minPost'];
+        var me = this;
+
+        tabs.forEach(function(tab) {
+            if (tab === activeTab) {
+                me.refs[tab].changeTabView(true);
+            } else {
+                me.refs[tab].changeTabView(false);
+            }
+        });
+    },
+
+    updateNavCreditCounts: function() {
+        var newCounts = [this.refs.spePost.getCreditCount(), 
+                         this.refs.majPost.getCreditCount(),
+                         this.refs.minPost.getCreditCount()];
+        this.refs.postNav.setState({creditCounts: newCounts},
+            this.updatePostStatus());
+    },
+
+    updatePostStatus: function() {
+
+        var newStatuses = [this.refs.spePost.setIfCompleted(), 
+                           this.refs.majPost.setIfCompleted(),
+                           this.refs.minPost.setIfCompleted()];
+        this.refs.postNav.setState({completed: newStatuses});
+    },
+
+    render: function() {
+        return (
+            <div id='check_my_post'>
+                <PostNav ref='postNav' 
+                         updateTab={this.changeActiveTab}
+                         getCreditCountClass={this.getCreditCountClass} />
+                <SpecialistPost ref='spePost' />
+                <MajorPost ref='majPost' />
+                <MinorPost ref='minPost' />
+            </div>
+        );
+    }
+});
+
+
+var PostNav = React.createClass({
+    getInitialState: function() {
+        return {
+            visible: this.getActiveTab() === '' ? 'spe' : this.getActiveTab(),
+            creditCounts: [0.0, 0.0, 0.0],
+            completed: [false, false, false]
+        }
+    },
+
+    getActiveTab: function() {
```

Comment:
Remove this function; save the value in a temporary variable in `getInitialState`.